tests/nosetests/pyanaconda_tests/modules/payloads/payload/module_payload_dnf_manager_test.py

```diff
@@ -918,3 +920,102 @@ def load_repository_test(self):
 
         repo.load.assert_called_once()
         self.assertEqual(repo.enabled, True)
+
+    def _create_repo(self, repo, repo_dir):
+        """Generate fake metadata for the repo."""
+        # Create the repodata directory.
+        os.makedirs(os.path.join(repo_dir, "repodata"))
+
+        # Create the repomd.xml file.
+        md_path = os.path.join(repo_dir, "repodata", "repomd.xml")
+        md_content = "Metadata for {}.".format(repo.id)
+
+        with open(md_path, 'w') as f:
+            f.write(md_content)
+
+        # Set up the baseurl.
+        repo.baseurl.append("file://" + repo_dir)
+
+    def load_no_repomd_hashes_test(self):
+        """Test the load_repomd_hashes method with no repositories."""
+        self.dnf_manager.load_repomd_hashes()
+        self.assertEqual(self.dnf_manager._md_hashes, {})
+
+    def load_one_repomd_hash_test(self):
+        """Test the load_repomd_hashes method with one repository."""
+        with TemporaryDirectory() as d:
+            r1 = self._add_repo("r1")
+            self._create_repo(r1, d)
+
+            self.dnf_manager.load_repomd_hashes()
+            self.assertEqual(self.dnf_manager._md_hashes, {
+                'r1': b"\x90\xa0\xb7\xce\xc2H\x85#\xa3\xfci"
+                      b"\x9e+\xf4\xe2\x19D\xbc\x9b'\xeb\xb7"
+                      b"\x90\x1d\xcey\xb3\xd4p\xc3\x1d\xfb",
```

Comment:
Doesn't really matter, but I'd go for all bytes escaped. This is perfectly subtly irregular, particularly with the line length matching. But I doubt anyone will actually look at it and try to compare.